src/extension.ts

```diff
@@ -92,10 +109,23 @@ async function bracePressed(editor: TextEditor): Promise<void> {
   }
 
   await editor.edit((edit: TextEditorEdit) => {
+    const config = workspace.getConfiguration("backticks");
+    const allowedFileExtensions = config.get<string[]>("useBraces") || [];
+    const currentFileExtension = document.fileName.substring(
+      document.fileName.lastIndexOf("."),
+    );
+
     for (let range of ranges) {
-      edit.replace(new Range(range.start, range.start.translate(0, 1)), "`");
-      if (range.end) {
-        edit.replace(new Range(range.end, range.end.translate(0, 1)), "`");
+      if (allowedFileExtensions.includes(currentFileExtension)) {
+        edit.replace(new Range(range.start, range.start.translate(0, 1)), "{`");
+        if (range.end) {
+          edit.replace(new Range(range.end, range.end.translate(0, 1)), "`}");
+        }
+      } else {
```

Comment:
This looks like it will use braces regardless of whether this is a JSX attribute or not or am I missing something? If so that's not ideal as then changed quotes outside of JSX become errors.